### PR TITLE
Set git diff to unicode

### PR DESCRIPTION
**Pull request by Moumoutaru**
Set's git diff encoding to Unicode

### CODE REVIEW

**Comment by willthames** [comment]
This looks good - are you able to add a unit test that fails before this change and succeeds afterwards? 

No worries if not, but I'd rather keep improving the minimal test suite as things crop up.

**Comment by Moumoutaru** [comment]
Makes sense. I will see what I can do.

**Comment by Moumoutaru** [comment]
@willthames Done and done.. It's a bit rough around the edges but seems to do the job. Open to suggestions.

**Comment by Moumoutaru** [comment]
@willthames Is there anything else you think should be done?

File: test/TestDiffEncoding.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    directory = os.path.dirname(__file__)
+    
+    def test_diff_encoding(self):
+        with io.open(os.path.join(self.directory, 'diff.txt'), 'r') as f:
+            try:
+                candidate = main.get_candidates_from_diff(f.read())
+                self.assertNotEqual(len(candidate), 1)
+            except TypeError as e:
+                self.assertRaises(TypeError)
```
Comment (willthames) [inline test/TestDiffEncoding.py]: do we expect to ever see this TypeError?

I'm not really sure what this bit is testing.

**Comment by willthames** [comment]
Thanks for the friendly nudge @Moumoutaru - I've had another look now. I think we're pretty close here.

File: test/TestDiffEncoding.py (hunk 1 above)
Comment (Moumoutaru) [reply test/TestDiffEncoding.py]: You are right in it's not being used.. Let me update it.

File: test/TestDiffEncoding.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        with io.open(os.path.join(self.directory, 'diff.txt'), 'r') as f:
+            for line in f.readlines():
+                encodedline = line.encode("utf-8")
+                difflines.append(encodedline)
+            try:
+                candidate = main.get_candidates_from_diff(difflines)
+                self.assertEqual(len(candidate), 1)
+            except UnicodeDecodeError as e:
+                self.assertFalse("False", e)
+        f.closed
```
Comment (willthames) [inline test/TestDiffEncoding.py]: I don't think this line is necessary

File: test/TestDiffEncoding.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    def test_diff_encoding(self):
+        difflines = []
+        with io.open(os.path.join(self.directory, 'diff.txt'), 'r') as f:
+            for line in f.readlines():
+                encodedline = line.encode("utf-8")
+                difflines.append(encodedline)
+            try:
+                candidate = main.get_candidates_from_diff(difflines)
+                self.assertEqual(len(candidate), 1)
+            except UnicodeDecodeError as e:
```
Comment (willthames) [inline test/TestDiffEncoding.py]: There's no need for the `try`/`except` block, as far as I can tell - the exception will be thrown if the test fails, and that will be handled properly by the test suite as a failing test

Comment (Moumoutaru) [reply test/TestDiffEncoding.py]: Your right. This isn't needed and the unit testing suite handles the error fine. I was thinking that handling it with a try catch the output would be cleaner but I suppose it really just obscures where the problem is. I'll update.

File: test/TestDiffEncoding.py (hunk 2 above)
Comment (Moumoutaru) [reply test/TestDiffEncoding.py]: Right, using `with` to open the file will handle closing it.

**Comment by Moumoutaru** [comment]
This has been updated with your suggestions.

**Comment by willthames** [comment]
Merged, thanks very much @Moumoutaru

**Comment by Moumoutaru** [comment]
Np!

**Pull request merged**